src/Product/VersionChecker.cs

```diff
@@ -0,0 +1,48 @@
+using System;
+using System.Linq;
+using System.Net.Http;
+using System.Reflection;
+using System.Text.Json.Serialization;
+
+namespace Azure.DataApiBuilder.Product;
+
+public static class VersionChecker
+{
+    private const string NuGetApiUrl = "https://api.nuget.org/v3-flatcontainer/Microsoft.DataApiBuilder/index.json";
+
+    public static void GetVersions(out string? latestVersion, out string? currentVersion)
+    {
+        latestVersion = FetchLatestNuGetVersion();
+        currentVersion = GetCurrentVersionFromAssembly(Assembly.GetExecutingAssembly());
+    }
+
+    private static string? FetchLatestNuGetVersion()
+    {
+        try
+        {
+            using HttpClient httpClient = new HttpClient { Timeout = TimeSpan.FromSeconds(2) };
+            NuGetVersionResponse? versionData = httpClient.GetFromJsonAsync<NuGetVersionResponse>(NuGetApiUrl)
+                .GetAwaiter().GetResult();
+
+            return versionData?.Versions
+                ?.Where(version => !version.Contains("-rc"))
+                .Max(); // Get the latest stable version

```

Comment:
Since Versions is an array of strings, would Max() return the version that has the higher number (instead of the one that comes last when ordered alphabetically)?

Here's what I mean: https://stackoverflow.com/questions/45582461/c-sharp-max-from-string-array-not-always-biggest-number

Reply:
Great catch, I used `Version` to resolve it.